Please enable yaml output for server reset-password

### TL;DR

`$ hcloud version`
hcloud 1.42.0

`$ hcloud server reset-password --help`
No option for YAML output :-(

Do I find the password in the server's description ... no: 
`$ hcloud server describe -o yaml SERVER_ID | grep -i pass`

I have to resort to parsing the output, e.g.
`$ hcloud server reset-password SERVER_ID | tail -n 1 | awk '{ print $NF }'`

This is too fragile.
If the output format ever changes, code breaks.


### Expected behavior

`$ hcloud server reset-password --output yaml SERVER_ID`

Thx :-)

Tried to test, but failed due to my non-existant GO skills:
Unfortunalety I cannot dive into this atm.
@phm07 maybe you could give me a quick hint to set up GO correctly.

I am on a Ubuntu 22.04 machine.
It has GO 1.18 installed in Debian/Ubuntu's standard locations, e.g. /usr/{lib,share,bin}/go-1.18

When I run `go install` as a regular user, it downloads hclould code in .`/go` and  I get:
```
$ go install github.com/hetznercloud/cli/cmd/hcloud@reset-password-output-options
go/pkg/mod/github.com/hetznercloud/cli@v1.42.1-0.20240319100245-975b1a8914bb/internal/cmd/iso/list.go:5:2: package slices is not in GOROOT (/usr/lib/go-1.18/src/slices)
```
I tried `export GOROOT='/usr/lib/go-1.18:/home/edge/go'` which did not help.
Reading the error message closer, I guess my GO installation is missing the `slices` package.
I will retry this evening and see how I can install that package.

--- a/internal/cmd/server/reset_password.go
+++ b/internal/cmd/server/reset_password.go
@@ -7,22 +7,28 @@ import (
 
 	"github.com/hetznercloud/cli/internal/cmd/base"
 	"github.com/hetznercloud/cli/internal/cmd/cmpl"
+	"github.com/hetznercloud/cli/internal/cmd/output"
+	"github.com/hetznercloud/cli/internal/cmd/util"
 	"github.com/hetznercloud/cli/internal/hcapi2"
 	"github.com/hetznercloud/cli/internal/state"
 )
 
 var ResetPasswordCmd = base.Cmd{
 	BaseCobraCommand: func(client hcapi2.Client) *cobra.Command {
-		return &cobra.Command{
-			Use:                   "reset-password <server>",
+		cmd := &cobra.Command{
+			Use:                   "reset-password [options] <server>",
 			Short:                 "Reset the root password of a server",
 			Args:                  cobra.ExactArgs(1),
 			ValidArgsFunction:     cmpl.SuggestArgs(cmpl.SuggestCandidatesF(client.Server().Names)),
 			TraverseChildren:      true,
 			DisableFlagsInUseLine: true,
 		}
+		output.AddFlag(cmd, output.OptionJSON(), output.OptionYAML())
+		return cmd
 	},
 	Run: func(s state.State, cmd *cobra.Command, args []string) error {
+		outputFlags := output.FlagsForCommand(cmd)
+
 		idOrName := args[0]
 		server, _, err := s.Client().Server().Get(s, idOrName)
 		if err != nil {
@@ -39,6 +45,16 @@ var ResetPasswordCmd = base.Cmd{
 
 		if err := s.ActionProgress(cmd, s, result.Action); err != nil {
 			return err
+		}
+
+		if outputFlags.IsSet("json") || outputFlags.IsSet("yaml") {
+			schema := make(map[string]interface{})
+			schema["root_password"] = result.RootPassword
+			if outputFlags.IsSet("json") {
+				return util.DescribeJSON(schema)
+			} else {
+				return util.DescribeYAML(schema)
+			}
 		}
 
 		cmd.Printf("Password of server %d reset to: %s\n", server.ID, result.RootPassword)

